src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -545,42 +546,61 @@ ucs_status_t uct_tcp_sockcm_ep_send(uct_tcp_sockcm_ep_t *cep)
     return uct_tcp_sockcm_ep_progress_send(cep);
 }
 
-static ucs_status_t uct_tcp_sockcm_ep_server_invoke_conn_req_cb(uct_tcp_sockcm_ep_t *cep)
+static ucs_status_t
+uct_tcp_sockcm_ep_server_invoke_conn_req_cb(uct_tcp_sockcm_ep_t *cep)
 {
-    uct_tcp_sockcm_priv_data_hdr_t *hdr     = (uct_tcp_sockcm_priv_data_hdr_t *)
-                                              cep->comm_ctx.buf;
-    struct sockaddr_storage remote_dev_addr = {0}; /* Suppress Clang false-positive */
+    uct_tcp_sockcm_priv_data_hdr_t *hdr = (uct_tcp_sockcm_priv_data_hdr_t*)
+                                                  cep->comm_ctx.buf;
+    struct sockaddr_storage saddr       = {0};
+    uct_tcp_device_addr_t *remote_dev_addr;
+    void *in_addr_ptr;
     uct_cm_listener_conn_request_args_t conn_req_args;
     char peer_str[UCS_SOCKADDR_STRING_LEN];
     char ifname_str[UCT_DEVICE_NAME_MAX];
     uct_cm_remote_data_t remote_data;
-    socklen_t remote_dev_addr_len;
+    socklen_t saddr_len;
     ucs_sock_addr_t client_saddr;
     ucs_status_t status;
+    size_t in_addr_len;
 
-    /* get the local interface name associated with the connected fd */
+    /* Get the local interface name associated with the connected fd */
     status = ucs_sockaddr_get_ifname(cep->fd, ifname_str, UCT_DEVICE_NAME_MAX);
     if (UCS_OK != status) {
         return status;
     }
 
-    /* get the device address of the remote peer associated with the connected fd */
-    status = ucs_socket_getpeername(cep->fd, &remote_dev_addr, &remote_dev_addr_len);
+    /* Get the device address of the remote peer associated with the connected
+     * fd */
+    status = ucs_socket_getpeername(cep->fd, &saddr, &saddr_len);
     if (status != UCS_OK) {
         return status;
     }
 
+    status = ucs_sockaddr_inet_addr_sizeof((struct sockaddr*)&saddr,
+                                           &in_addr_len);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    remote_dev_addr = ucs_alloca(sizeof(*remote_dev_addr) + in_addr_len);
+    in_addr_ptr     = (void*)(remote_dev_addr + 1);
+
+    remote_dev_addr->flags     = 0u;
+    remote_dev_addr->sa_family = saddr.ss_family;
+    memcpy(in_addr_ptr, ucs_sockaddr_get_inet_addr((struct sockaddr*)&saddr),
+           in_addr_len);
+
     remote_data.field_mask            = UCT_CM_REMOTE_DATA_FIELD_DEV_ADDR        |
                                         UCT_CM_REMOTE_DATA_FIELD_DEV_ADDR_LENGTH |
                                         UCT_CM_REMOTE_DATA_FIELD_CONN_PRIV_DATA  |
                                         UCT_CM_REMOTE_DATA_FIELD_CONN_PRIV_DATA_LENGTH;
-    remote_data.dev_addr              = (uct_device_addr_t *)&remote_dev_addr;
-    remote_data.dev_addr_length       = remote_dev_addr_len;
+    remote_data.dev_addr              = (uct_device_addr_t*)remote_dev_addr;
+    remote_data.dev_addr_length       = sizeof(remote_dev_addr) + in_addr_len;
```

Comment:
yes, in TCP SOCKCM.
no need to pack system ns information. they are 100% reachable, since `ucs_socket_getpeername()` completed with success.
we just always pack in_addr of loopback.